core/gate.cc

```diff
@@ -72,7 +72,7 @@ int Gate::AddHook(GateHook *hook) {
   }
 
   hooks_.push_back(hook);
-  std::sort(hooks_.begin(), hooks_.end(), GateHookComp);
+  std::sort(hooks_.begin(), hooks_.end());
```

Comment:
(see later comment about sort-right-after-insert ... I wound up going through this backwards)

Reply:
Good point. Switched to `std::set` over `std::vector` here since this isn't on the fast path.